src/Api/Client.php

```diff
@@ -127,10 +124,43 @@ public function call($endPoint, $httpRequestMethod, $content = [])
 		return $response;
 	}
 
+    /**
+     * @param string
+     * @return resource|false
+     */
+    private function getCurlHandle($endPoint)
+    {
+        $url = self::URL.$endPoint;
+        if (array_key_exists($endPoint, $this->curlHandles)) {
+            if (!isset($this->curlHandles[$endPoint])) {
+                $this->curlHandles[$endPoint] = curl_init($url);
+            }
+            
+            return $this->curlHandles[$endPoint];
+        }
+        
+        return false;
+    }
+	
+	public function closeAll() {
+	    try {
+	        foreach ($this->curlHandles as $key => $curlHandle) {
+	            if (!is_null($curlHandle)) {
+	                curl_close($curlHandle);
+                }
+	            
+	            if (!is_null($this->curlHandles[$key])) {
```

Comment:
I would assume that the second if will reach or?

Reply:
if you can tell me for sure, that the variables generated by the foreach is a pointer to the content of the array I would only need one, otherwise I assume I copy the connection and have to close copy and original.

Reply:
I am not 100 percent sure but it should be a pointer. Otherwise, foreach loops would be really useless. But exactly for this case I thought to keep the second if because you are there just using the key =)

Reply:
done